language/move-binary-format/src/compatibility.rs

```diff
@@ -268,3 +268,89 @@ fn type_parameter_phantom_decl_compatibile(
     // old_type_paramter.is_phantom => new_type_parameter.is_phantom
     !old_type_parameter.is_phantom || new_type_parameter.is_phantom
 }
+
+/// A simpler, and stricter compatibility checker relating to the inclusion of the old module in
+/// the new.
+#[derive(Clone, Debug, Ord, PartialOrd, Eq, PartialEq)]
+pub enum InclusionCheck {
+    Subset,
+    Equal,
+}
+
+impl InclusionCheck {
+    pub fn check(&self, old_module: &Module, new_module: &Module) -> PartialVMResult<()> {
```

Comment:
Nit: I assume that we're checking the `old_module` is a subset or equal to `new_module`, but we should document that explicitly.